main.py

```diff
@@ -37,4 +38,18 @@ def maybe_chdir():
         maybe_chdir()
     server_name = "0.0.0.0" if args.listen else None
     import src.common_ui
-    src.common_ui.on_ui_tabs().launch(share=args.share, server_name=server_name)
+
+    ui_block = src.common_ui.on_ui_tabs()
+    
+    if args.api is not True:
+        ui_block.launch(share=args.share, server_name=server_name)
+    else:
+        app, _, _ = ui_block.launch(share=args.share, server_name=server_name, prevent_thread_lock=True)
```

Comment:
What is prevent_thread_lock? About locks: core_generation_funnel is not "multithread-safe" - at most one thread can yield from it at the same time - otherwise _everything_ breaks.

Reply:
This isn't about multithreading this is about blocking the main thread. Without it no code below is executed. In order to attach the api to the same port as the webui you need to have code afterwards. This option is also enabled in auto1111  with the --api option and it doesn't cause any issues with this extension (https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/webui.py).

My understanding is almost every python webserver is single threaded and the strategy with python is usually to just run multiple single threaded apps as workers rather than multi-threading.